pymysqlreplication/binlogstream.py

```diff
@@ -367,7 +367,7 @@ def __connect_to_stream(self):
             # master_heartbeat_period is nanoseconds
             heartbeat = int(heartbeat * 1000000000)
             cur = self._stream_connection.cursor()
-            cur.execute("SET @master_heartbeat_period= %d" % heartbeat)
+            cur.execute(f"SET @master_heartbeat_period= {heartbeat}")
```

Comment:
This is not the recommended way to use DB API in Python. First argument of `execute` must be a query string with placeholders %s for the following arguments to be replaced with.

https://peps.python.org/pep-0249/#id20

Reply:
> This is not the recommended way to use DB API in Python. First argument of `execute` must be a query string with placeholders %s for the following arguments to be replaced with.
> 
> https://peps.python.org/pep-0249/#id20

thanks for a good point! if were not you I could not have recognized this recommandation